astropy/convolution/convolve.py

```diff
@@ -22,7 +23,10 @@
 LIBRARY_PATH = os.path.dirname(__file__)
 
 try:
-    _convolve = load_library("_convolve", LIBRARY_PATH)
+    with warnings.catch_warnings():
+        # distutils.sysconfig module is deprecated in Python 3.10
+        warnings.simplefilter('ignore', DeprecationWarning)
+        _convolve = load_library("_convolve", LIBRARY_PATH)
```

Comment:
This fix does not look like it is going to work when the deprecated feature is removed. Is there no replacement for this?

Reply:
Numpy has some plan : https://github.com/numpy/numpy/issues/18588
But here we are using a smaller helper function that relies on distutils to get the shared library extension on different platforms. We could replace this with some custom code I guess.

Reply:
So, instead of merging this, which would completely mask the real problem, should we wait a bit and see? Not like we can test against Python 3.10 in CI right now anyway...

Reply:
I don't think they will fix this in a few weeks, and we/they have time until 3.12
Actions seems to support 3.10b4 now so I'm trying to add a job.

Reply:
Okay, as long as we open follow-up issue about this, I guess I can live with this for now.

Reply:
Yes, hopefully this can be an "upstream fix required"!